aiosmtpd/docs/smtp.rst

```diff
@@ -104,31 +107,82 @@ override to provide additional responses.
 Handler hooks
 =============
 
-Handlers can implement a few hooks that get called during the ``SMTP`` dialog,
-or in exceptional cases.  More hooks may be added in the future, but for now
-the following handler hooks are defined:
+Handlers can implement a hooks that get called during the ``SMTP`` dialog, or
+in exceptional cases.  These *handler hooks* are all called asynchronously and
+they *must* return a status string, since the default statuses are *not*
+returned when a hook is defined.  Individual handlers may have additional
+responsibilities to replace default behavior, as described below.
```

Comment:
I suggest changing this paragraph to:

```
Handlers can implement hooks that get called during the ``SMTP`` dialog, or in
exceptional cases.  These *handler hooks* are all called asynchronously and
they *must* return a status string such as ``'250 OK'``.  All handlers are
optional and have certain default behaviors when omitted, but when handlers are
not omitted they may have additional responsibilities, as described below.
```

This fixes a typo ("a hooks") and tries to clarify two phrases that confused me on a first reading: "since the default statuses are not returned when a hook is defined" and "to replace default behavior". I think it is important to highlight that all handler hooks are optional.